server/middleware/fetchFrontendComponentMiddleware.ts

```diff
@@ -0,0 +1,28 @@
+import { Router } from 'express'
+import { DataAccess } from '../data'
+import logger from '../../logger'
+import config from '../config'
+
+export default function setUpFrontendComponents({ frontendComponentApiClient }: DataAccess): Router {
+  const router = Router({ mergeParams: true })
+
+  router.get('*', async (req, res, next) => {
+    try {
+      // Frontend components API is only used when feature toggle is provided
+      if (config.featureToggles.frontendComponentsApiToggleEnabled) {
+        const { user } = res.locals
+        const [footer] = await Promise.all([frontendComponentApiClient.getComponent('footer', user.token)])
+        res.locals.feComponents = {
+          footer: footer.html,
+          cssIncludes: [...footer.css],
+          jsIncludes: [...footer.javascript],
+        }
```

Comment:
I wonder if this object type could be defined as a type in our viewModels module? 🤔 
Something like
```
declare module `viewModels` {
...
  export interface FrontendComponentsPageAdditions {
    footerHtml: string,
    cssIncludes: Array<string>,
    jsIncludes: Array<string>,
  }
...

```

Reply:
Then what you've written here becomes:
```
        res.locals.feComponents = {
          footerHtml: footer.html,
          cssIncludes: [...footer.css],
          jsIncludes: [...footer.javascript],
        } as FrontendComponentsPageAdditions
```

Is there much gain from this? Well, arguably not; but we are using typescript so I like to use types where we can 😁 . Plus it allows us to refactor this out into a mapper function or similar (I'm not proposing you do that here, but it does support us doing that in the future if we want)

Reply:
Used types here following the change for adding `export interface FrontendComponentsPageAdditions`